metrics/ci_integrations/test/source/buildkite/config/validator/buildkite_source_validator_test.dart

```diff
@@ -0,0 +1,359 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:ci_integration/client/buildkite/models/buildkite_token.dart';
+import 'package:ci_integration/client/buildkite/models/buildkite_token_scope.dart';
+import 'package:ci_integration/integration/validation/model/field_validation_result.dart';
+import 'package:ci_integration/integration/validation/model/validation_result_builder.dart';
+import 'package:ci_integration/source/buildkite/config/model/buildkite_source_config.dart';
+import 'package:ci_integration/source/buildkite/config/model/buildkite_source_config_field.dart';
+import 'package:ci_integration/source/buildkite/config/validation_delegate/buildkite_source_validation_delegate.dart';
+import 'package:ci_integration/source/buildkite/config/validator/buildkite_source_validator.dart';
+import 'package:ci_integration/source/buildkite/strings/buildkite_strings.dart';
+import 'package:ci_integration/util/authorization/authorization.dart';
+import 'package:ci_integration/util/model/interaction_result.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../../../test_utils/extensions/interaction_result_answer.dart';
+
+void main() {
+  group("BuildkiteSourceValidator", () {
+    const accessToken = 'token';
+    const organizationSlug = 'organization';
+    const pipelineSlug = 'pipeline';
+    const message = 'message';
+
+    final config = BuildkiteSourceConfig(
+      accessToken: accessToken,
+      organizationSlug: organizationSlug,
+      pipelineSlug: pipelineSlug,
+    );
+
+    final auth = BearerAuthorization(accessToken);
+
+    const allRequiredScopesToken = BuildkiteToken(
+      scopes: [
+        BuildkiteTokenScope.readOrganizations,
+        BuildkiteTokenScope.readPipelines,
+      ],
+    );
+    const pipelineScopeToken = BuildkiteToken(
+      scopes: [BuildkiteTokenScope.readPipelines],
+    );
+    const organizationScopeToken = BuildkiteToken(
+      scopes: [BuildkiteTokenScope.readOrganizations],
+    );
+
+    final validationDelegate = _BuildkiteSourceValidationDelegateMock();
+    final validationResultBuilder = _ValidationResultBuilderMock();
+    final validator = BuildkiteSourceValidator(
+      validationDelegate,
+      validationResultBuilder,
+    );
+
+    tearDown(() {
```

Comment:
Would it be great to move the `tearDown` method to be right above the test cases?